### PR TITLE
[Pytest conversion] test_warehouses

### PR DESCRIPTION
**Pull request by McKnight-42**
resolves: continued pytest conversions.

### Description

migrating snowflake specific integration test to a new functional pytest

### Checklist

- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have updated the `CHANGELOG.md` and added information about my change to the "dbt-snowflake next" section.

### CODE REVIEW

File: tests/functional/warehouse_test/fixtures.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+"""
+
+
+@pytest.fixture(scope="class")
+def models():
+    return {
+        "override_warehouse.sql": models__override_warehouse_sql,
+        "expected_warehouse.sql": models__expected_warehouse_sql,
+        "invalid_warehouse.sql": models__invalid_warehouse_sql,
+    }
```
Comment (emmyoop) [inline tests/functional/warehouse_test/fixtures.py:42]: This should actually be defined in your `test_warehouses.py`, not here.  That way we can easily see what `sql` files make up the models.  Same for the `project_config_models`

File: tests/functional/warehouse_test/fixtures.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+
+@pytest.fixture(scope="class")
+def project_files(
+    project_root,
+    models,
+    project_config_models,
+):
+    write_project_files(project_root, "models", models)
+    write_project_files(project_root, "project-config-models", project_config_models)
```
Comment (emmyoop) [inline tests/functional/warehouse_test/fixtures.py:60]: This is also unnecessary when you define the models in your tests.  This is done when we're setting up the project(https://github.com/dbt-labs/dbt-core/blob/1071a4681df91633301fdf23e34de819b66fbeb7/core/dbt/tests/fixtures/project.py#L331-L338).

File: tests/functional/warehouse_test/test_warehouses.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import os
+
+
+class TestModelWarehouse():
+    def dir(value):
+        return os.path.normpath(value)
+
+    @pytest.fixture(scope="class")
+    def model_path(self):
+        return self.dir('models')
```
Comment (emmyoop) [inline tests/functional/warehouse_test/test_warehouses.py:20]: Is this pattern something that needs to be done for snowflake?  It's just not a pattern I've seen in our tests yet but I haven't worked with the snowflake tests either.

**Review by emmyoop — COMMENTED** [review]
I would remove the fixture file completely and just put it all into your test file.  It's the pattern most of the tests follow for fixtures that are not shared across files.

**Comment by ChenyuLInx** [comment]
The fixture thing is probably the fault of my conversion script, since I just did it as a standard thing to do. Good call out @emmyoop

**Review by dbeatty10 — COMMENTED** [review]
I'm probably overlooking this somehow -- how is this test only running against the Snowflake test profile and not running against other database adapters?

**Comment by McKnight-42** [comment]
> 
@dbeatty10 there are a few integration/functional tests that fall under very niche use cases so we have some tests that only appear in the adapters that need them. this being one of them that is literally only used for snowflake so there is no dbt-core version of it to be inherited (is this something we should move away from, who knows).

File: tests/functional/warehouse_test/test_warehouses.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+"""
+
+
+@pytest.fixture(scope="class")
+def models():
+    return {
+        "override_warehouse.sql": models__override_warehouse_sql,
+        "expected_warehouse.sql": models__expected_warehouse_sql,
+        "invalid_warehouse.sql": models__invalid_warehouse_sql,
+    }
```
Comment (emmyoop) [inline tests/functional/warehouse_test/test_warehouses.py:47]: Move this inside `TestModelWarehouse`

File: tests/functional/warehouse_test/test_warehouses.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+
+@pytest.fixture(scope="class")
+def project_config_models():
+    return {
+        "override_warehouse.sql": project_config_models__override_warehouse_sql,
+        "expected_warehouse.sql": project_config_models__expected_warehouse_sql,
+    }
```
Comment (emmyoop) [inline tests/functional/warehouse_test/test_warehouses.py:55]: Move this inside `TestConfigWarehouse`.  It makes the discoverability of what models are used in what tests much higher.

**Review by emmyoop — CHANGES_REQUESTED** [review]

File: tests/functional/warehouse_test/test_warehouses.py (hunk 5 above)
Comment (McKnight-42) [reply tests/functional/warehouse_test/test_warehouses.py]: done, had to rename `project_configt_models` to `models` once I moved it into the class but still passes locally

File: tests/functional/warehouse_test/test_warehouses.py (hunk 4 above)
Comment (McKnight-42) [reply tests/functional/warehouse_test/test_warehouses.py]: done!

File: tests/functional/warehouse_test/test_warehouses.py [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+class TestModelWarehouse:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "override_warehouse.sql": models__override_warehouse_sql,
+            "expected_warehouse.sql": models__expected_warehouse_sql,
+            "invalid_warehouse.sql": models__invalid_warehouse_sql,
+        }
+    def test_snowflake_override_ok(self, project):
```
Comment (emmyoop) [inline tests/functional/warehouse_test/test_warehouses.py:49]:
```suggestion
        }

    def test_snowflake_override_ok(self, project):
```

**Review by emmyoop — APPROVED** [review]
Just a tiny formatting suggestions then this looks good to merge in!

**Pull request merged**